merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/TaskId.java

```diff
@@ -1,10 +1,11 @@
 package gov.nasa.jpl.aerie.merlin.driver.engine;
 
-import java.util.UUID;
+import java.util.Random;
 
 /** A typed wrapper for task IDs. */
-/*package-local*/ record TaskId(String id) {
+/*package-local*/ record TaskId(long id) {
+  private static final Random r = new Random();
   public static TaskId generate() {
-    return new TaskId(UUID.randomUUID().toString());
+    return new TaskId(r.nextLong());
```

Comment:
A `long` is 64 bits and a `UUID` is 128 bits (comprised of a LSB and MSB `long`). I think a `UUID` should be used here out of an abundance of caution but maybe that's silly since 64 bits should make duplicates rare anyways.

Reply:
Looks like there's also a [`UUID` PG type](https://www.postgresql.org/docs/9.1/datatype-uuid.html).

Reply:
Since IDs are being assigned sequentially here, duplicates should be more than just rare -- it will take exactly 2**64 tasks before we see the same ID in the same context (per-simulation, it looks like here).

I think Postgres also finds it easier to maintain ordered indices on numbers, especially densely packed numbers (vs. the sparse cover for random UUIDs).

Reply:
Maybe I'm missing something but it looks like the IDs are being assigned with a pseudorandom distribution (`Random.nextLong()`) and not sequentially (`lastId+1`)?

Reply:
Nope, I'm the one missing something! I misinterpreted the code by looking at only the diff.

(that said, it'd be good to know if/why we can't use sequential task IDs, for the reasons above)

Reply:
Good point. I'm onboard with using sequential IDs here

Reply:
Two points:

1. On the randomeness of Random.nextLong(), I did some research on the chance of collision, and as discussed above, it's certainly random enough for us to not be concerned.

2. I'm also fine with using sequential IDs if we really want to, but we would have to verify the ID is unique every time we generate one, which is extra work we don't have to do when generating random IDs, and I'm not sure how much work it would take to include enough context to check that when generating the IDs. Furthermore, it may be worth noting that though the task IDs do become activity instance IDs after simulation, they are not stored in Postgres except for Directive IDs, which are not generated in the above code. When stored in postgres, sequential IDs are automatically generated. When querying for simulated activities through our system, the ID for each activity is either its directive ID (if present) or a sequentially generated long (see `liftDirectiveIDs()` in the PostgresResultsCellRepository in merlin-server).

Reply:
One thing I remember doing with `Random` was to use `new Random(System.currentTimeMillis())` to update the seed for random. Not sure if you want to do it here?

Reply:
I think this discussion is relevant to the child activity ID collision issue that @camargo brought up - we're re-using IDs from different places (Planned activity instance IDs reused for TaskIds and simulated activity IDs), which I think causes some unnecessary coupling and confusion.

I think I would advocate for keeping ActivityInstanceIds, TaskIds, and SimulatedActivityIds completely separate, and tracking mappings between them as necessary. This way each of those can be generated by a simple counter (either in postgres or in merlin, depending on which ID we're talking about), and we can guarantee no collisions.

I admit I lack the intuition about collision likelihood - any non-zero probability makes me think we need to handle the case where a collision does happen... though clearly with UUID standard practice is to assume they'll never happen 🤔

Reply:
> Furthermore, it may be worth noting that though the task IDs do become activity instance IDs after simulation, they are not stored in Postgres except for Directive IDs, which are not generated in the above code. [...] When querying for simulated activities through our system, the ID for each activity is either its directive ID (if present) or a sequentially generated long (see `liftDirectiveIDs()` in the PostgresResultsCellRepository in merlin-server).

@kroffo reading what you said more carefully, it occurs to me that we don't need to change TaskId to use a Long... right? Can we just continue using UUIDs?

Reply:
Let me point your attention to line 405 of SimulationDriver.java.

If a simulated activity instance does not have an associated activity instance ID, then the Task ID used for it during simulation becomes its activity instance ID. So pray tell, how would you convert a `UUID` to a `long`?

Reply:
After some discussion, @mattdailis and I determined that because of the line I pointed out Task IDs are really being used for two purposes. I'm going to reduce this by computing IDs for simulated activities sequentially instead of using the TaskID, as is done in the PostgresResultsCellRepository for solving this same problem.

Reply:
This issue has now been resolved. All changes associated with the TaskID change have been removed. Reviewers should examine the changes from lines 403-417 of SimulationEngine.java as these are new since the latest update.